tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -63,6 +63,134 @@ public function test_wp_enqueue_script() {
+		wp_add_inline_script( 'ds-i3-1', 'console.log("before one");', 'before' );
+		$output   = get_echo( 'wp_print_scripts' );
+		$expected  = "<script type='text/javascript' id='ds-i3-1-js-before'>\n";
+		$expected .= "console.log(\"before one\");\n";
+		$expected .= "</script>\n";
+		$expected .= "<script type='text/javascript' src='http://example.org/ds-i3-1.js' id='ds-i3-1-js' defer></script>\n";
+		$expected .= "<script type='text/javascript' src='http://example.org/ds-i3-2.js' id='ds-i3-2-js' defer></script>\n";
+		$expected .= "<script type='text/javascript' src='http://example.org/ms-i3-1.js' id='ms-i3-1-js' defer></script>\n";
+		array_push( $data, array( $expected, $output, 'Top most has before inline script. All the script in the chain defer.' ) );
+
+		// If there are two dependencies chain; rules are applied to the scripts in the chain having a `before` inline script.
+		wp_enqueue_script( 'ch1-ds-i4-1', 'http://example.org/ch1-ds-i4-1.js', array(), null, array( 'strategy' => 'defer' ) );
+		wp_enqueue_script( 'ch1-ds-i4-2', 'http://example.org/ch1-ds-i4-2.js', array( 'ch1-ds-i4-1' ), null, array( 'strategy' => 'defer' ) );
+		wp_enqueue_script( 'ch2-ds-i4-1', 'http://example.org/ch2-ds-i4-1.js', array(), null, array( 'strategy' => 'defer' ) );
+		wp_enqueue_script( 'ch2-ds-i4-2', 'http://example.org/ch2-ds-i4-2.js', array( 'ch2-ds-i4-1' ), null, array( 'strategy' => 'defer' ) );
+		wp_add_inline_script( 'ch2-ds-i4-2', 'console.log("before one");', 'before' );
+		wp_enqueue_script( 'ms-i4-1', 'http://example.org/ms-i4-1.js', array( 'ch2-ds-i4-1', 'ch2-ds-i4-2' ), null, array( 'strategy' => 'defer' ) );
+		$output   = get_echo( 'wp_print_scripts' );
+		$expected  = "<script type='text/javascript' src='http://example.org/ch1-ds-i4-1.js' id='ch1-ds-i4-1-js' defer></script>\n";
+		$expected .= "<script type='text/javascript' src='http://example.org/ch1-ds-i4-2.js' id='ch1-ds-i4-2-js' defer></script>\n";
+		$expected .= "<script type='text/javascript' src='http://example.org/ch2-ds-i4-1.js' id='ch2-ds-i4-1-js'></script>\n";
+		$expected .= "<script type='text/javascript' id='ch2-ds-i4-2-js-before'>\n";
+		$expected .= "console.log(\"before one\");\n";
+		$expected .= "</script>\n";
+		$expected .= "<script type='text/javascript' src='http://example.org/ch2-ds-i4-2.js' id='ch2-ds-i4-2-js' defer></script>\n";
+		$expected .= "<script type='text/javascript' src='http://example.org/ms-i4-1.js' id='ms-i4-1-js' defer></script>\n";
+		array_push( $data, array( $expected, $output, 'Only top dependency script in chain two should be blocking.' ) );
+
+		return $data;
+	}
+
+	/**
+	 * Test standalone tests.
+	 *
+	 * @ticket 12009
+	 * @dataProvider data_standalone_inline_script
+	 */
+	public function test_standalone_inline_script( $expected, $output, $message ) {
+		$this->assertSame( $expected, $output, $message );
+	}
+
+	public function data_standalone_inline_script() {
```

Comment:
Missing docblock, I know it's a data provider for the test but we should still have docblocks or not?

Reply:
Same comment about the use of a data provider here.

Reply:
As mentioned above this will be handled in PR #50.